eras/conway/impl/src/Cardano/Ledger/Conway/PParams.hs

```diff
@@ -677,7 +678,7 @@ updateField = \case
   11 -> field (\x up -> up {cppTau = THKD (SJust x)}) From
   16 -> field (\x up -> up {cppMinPoolCost = THKD (SJust x)}) From
   17 -> field (\x up -> up {cppCoinsPerUTxOByte = THKD (SJust x)}) From
-  18 -> field (\x up -> up {cppCostModels = THKD (SJust x)}) From
+  18 -> field (\x up -> up {cppCostModels = THKD (SJust x)}) $ D decodeCostModelsFailingOnError
```

Comment:
This is an incorrect function to use, because `decodeCostModelsFailingOnError` will fail for unknown languages, which is not what we want.

We want PParamsUpdate to fail whenever there is a CostModel for a known language that is invalid, but it should still succeed whenever the CostModel is for a language that we don't know about. 

In particular this is the line that is wrong for this decoder, but it is ok for PParams: https://github.com/input-output-hk/cardano-ledger/blob/c8874b4ae6ef569d28f492dc6d948bfdc607a3ef/libs/cardano-ledger-core/src/Cardano/Ledger/Plutus/CostModels.hs#L341

Reply:
You're totally right, I completely missed half of the point of the task, sorry about that. 
Writing another function that is a combination of the two,  `decodeCostModelsFailingOnError` and `decodeCostModelsCollectingErrors` - that fails for invalid but collects for unknown.

Reply:
"Incidentally", the part I didn't write tests for